### PR TITLE
Develop branch prebuilt frontend bundle (#1576)

### LINKED ISSUE #1576: Prebuild for develop branch
### Description

Currently, to install directly from develop branch, the frontend will need to be built on the host machine. This will take some time on the host machine to build the frontend bundle. By rebuilding the frontend bundle, the host machine will not need to build these packages.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #1576

### CODE REVIEW

File: .github/workflows/prebuild.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          path: |
+            taipy/gui/webapp
+            taipy/gui_core/lib
+          key: frontend-build-${{ runner.os }}-${{ steps.hash-frontend.outputs.HASH }}
+
+      - name: Setup node ${{ matrix.node-version }} on ${{ matrix.os }}
+        if: steps.cache-fe-build.outputs.cache-hit != 'true'
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (FredLL-Avaiga) [inline .github/workflows/prebuild.yml]: I don't see any matrix defined and I suppose we don't need one here ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

**Review by FabienLelaquais — DISMISSED** [review]
Looks really nice!

**Comment by jrobinAV** [comment]
There is a failing action. 
https://github.com/Avaiga/taipy/actions/runs/10072401611/job/27844607690?pr=1577
Note that all the tests are good, but after running the tests, the action just wait for the timeout to cancel it.
It happened multiple times this week on other PRs on the same configuration (3.8, windows-latest, min). After multiple retries it worked. If you have any idea on how to investigate the issue...

File: .github/workflows/prebuild.yml (hunk 1 above)
Comment (dinhlongviolin1) [reply .github/workflows/prebuild.yml]: oops, thanks for finding it! you are the best!

**Review by joaoandre-avaiga — DISMISSED** [review]

**Review by jrobinAV — DISMISSED** [review]

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: .github/workflows/prebuild.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Restore cached frontend build
+        id: cache-fe-build
+        uses: actions/cache@v4
+        with:
+          path: |
+            taipy/gui/webapp
+            taipy/gui_core/lib
+          key: frontend-build-${{ runner.os }}-${{ steps.hash-frontend.outputs.HASH }}
+
+      - name: Setup node ${{ matrix.node-version }} on ${{ matrix.os }}
```
Comment (FredLL-Avaiga) [inline .github/workflows/prebuild.yml]: still ref to matrix ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
the matrix is there

File: .github/workflows/prebuild.yml (hunk 2 above)
Comment (dinhlongviolin1) [reply .github/workflows/prebuild.yml]: somehow i missed the search again. Thanks!

**Review by FredLL-Avaiga — APPROVED** [review]
🥇

**Pull request merged**